File: Mathematics/02-GCD_EuclidsDiv.cpp
#include <iostream>
#include <vector>
using namespace std;
int main() {
  int a, b;
  cin >> a >> b;

  int maxm = max(a, b);
  int minm = min(a, b);

  while (maxm % minm != 0) {
    int temp = maxm;
    maxm = minm;
    minm = temp % minm;
  }

  cout << minm << endl;
  return 0;
}

File: Mathematics/02-GCD_EuclidsDiv.js
var findGCD = function (arr) {
  let maxm = arr[0];
  let minm = arr[0];

  for (let i = 1; i < arr.length; i++) {
    maxm = Math.max(maxm, arr[i]);
    minm = Math.min(minm, arr[i]);
  }

  if (minm == 0) {
    return maxm;
  }

  // Euclidean method
  while (minm > 0) {
    let remainder = maxm % minm;
    maxm = minm;
    minm = remainder;
  }

  return maxm;
};
